compiler/src/iree/compiler/Codegen/Common/GPU/GPUConvertToCoalescedDMA.cpp

```diff
@@ -212,6 +213,15 @@ struct ConvertGatherOpToCoalescedDMA
     // Cast indices to index type if needed
     Value indexIndices = castIndicesToIndexType(rewriter, loc, indicesWgSlice);
 
+    auto indicesTensorType = cast<RankedTensorType>(indexIndices.getType());
```

Comment:
... shouldn't this happen alongside all the other tensor=>vector conversions? I'd expect there to be a pass later in the IR for that.

Reply:
Can you link that PR?

Reply:
Agree with Krzysztof, we should be doing the tensor->vector conversion as a part of vectorization. This is a premature lowering, can you share why you needed to do this?